Splitter/Splitter/Controllers/MyBillsViewController.swift

```diff
@@ -8,20 +8,129 @@
         
-        view.backgroundColor = .white
+        setupHierarchy()
+        setupViews()
+        setupLayout()
+    }
+    
+    private func setupHierarchy() {
+        view.addSubview(carousel)
         view.addSubview(titleLabel)
+        view.addSubview(addButton)
+        view.addSubview(deleteButton)
+        view.addSubview(noBillsLabel)
+
+    }
+    
+    private func setupViews() {
+        view.backgroundColor = Color.mainBackground
+        
+        carousel.dataSource = carouselDatasource
+        carousel.delegate = carouselDelegate
+        carousel.isPagingEnabled = true
+        carousel.type = .coverFlow
+        carousel.backgroundColor = .clear
+        
+        titleLabel.text = Localized.myBillsViewControllerTitle
+        
+        addButton.addTarget(self, action: #selector(addButtonWasTapped), for: .touchUpInside)
+        
+        deleteButton.isHidden = true
+        
+        noBillsLabel.text = Localized.noBillsMessage
+        noBillsLabel.isHidden = true
+    }
+    
+    private func setupLayout() {
+        carousel.pinToSuperviewEdges()
+        
+        titleLabel.pinToSuperview(edges: [.left, .right])
+        titleLabel.pinTop(to: view,
+                          constant: Layout.titleLabelY,
+                          priority: .required,
+                          relatedBy: .equal)
+        
+        addButton.pinToSuperview(edges: [.left, .bottom])
+        
+        deleteButton.pinToSuperview(edges: [.right, .bottom])
+        
+        noBillsLabel.pinToSuperview(edges: [.left, .right])
+        noBillsLabel.centerXToSuperview()
+        noBillsLabel.centerYToSuperview()
+    }
+    
+    private func getUserBills() {
+        let firebaseData = FirebaseData()
+        firebaseData.findBillsWith(userID: currentUser.id,
+                                   completion: { bills in
+            if let userbills = bills {
+                self.userBills = userbills
+                self.hideNoBillsLabel()
+                self.showDeleteButton()
+            } else {
+                self.showNoBillsLabel()
+                self.hideDeleteButton()
+            }
+        })
+    }
+    
+    private func showDeleteButton() {
+        DispatchQueue.main.async { [weak self] in
+            self?.deleteButton.isHidden = false
+        }
+    }
+    
+    private func hideDeleteButton() {
+        DispatchQueue.main.async { [weak self] in
+            self?.deleteButton.isHidden = true
+        }
+    }
+    
+    private func showNoBillsLabel() {
+        DispatchQueue.main.async { [weak self] in
+            self?.noBillsLabel.isHidden = false
+        }
+    }
+    
+    private func hideNoBillsLabel() {
+        DispatchQueue.main.async { [weak self] in
+            self?.noBillsLabel.isHidden = true
+        }
+    }
+    
+    @objc private func addButtonWasTapped() {
+        let newBillViewController = NewBillViewController()
+        newBillViewController.currentUserID = currentUser.id
```

Comment:
do you only create the new bill view controller in code? 

If you do, you could perhaps make the `currentUserID` a initialiser parameter. This has the benefit of a) then being a `let` variable, and b) makes it more obvious that this class can't function without a currentUser. 

Whilst setting variables on a viewcontroller is a _very_ common pattern on iOS, it is a bit of an antipattern. The reason being that I have to know/remember to set the variable, but as it is just a random variable, I'm given no information that it is required. If it is an initialiser param, then I am given no choice but to add it - highlighting to the consumer of the class that it is a requirement. The posh name for this is "[connascence by algorithm](https://en.wikipedia.org/wiki/Connascence_(computer_programming)#Connascence_of_Algorithm_.28CoA.29)" - you have to know how the class functions in order to be able to use it.

Reply:
This is great! Thanks. I dont know why i never thought of using an initialiser for the viewcontroller. Seems ridiculous now. :)

Reply:
Done e9e96da7da9fa6c56336e41714f78e8f55f067fd